File: src/main/java/com/patres/timetable/service/dto/SubjectDTO.java
package com.patres.timetable.service.dto;


import javax.validation.constraints.*;
import java.io.Serializable;
import java.util.Objects;

public class SubjectDTO extends AbstractDivisionOwnerDTO implements Serializable {

    @NotNull
    private String name;

    private String shortName;

    private String colorBackground;

    private String colorText;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getShortName() {
        return shortName;
    }

    public void setShortName(String shortName) {
        this.shortName = shortName;
    }

    public String getColorBackground() {
        return colorBackground;
    }

    public void setColorBackground(String colorBackground) {
        this.colorBackground = colorBackground;
    }

    public String getColorText() {
        return colorText;
    }

    public void setColorText(String colorText) {
        this.colorText = colorText;
    }

    @Override
    public String toString() {
        return "SubjectDTO{" +
            "id=" + getId() +
            ", name='" + getName() + "'" +
            ", shortName='" + getShortName() + "'" +
            ", colorBackground='" + getColorBackground() + "'" +
            ", colorText='" + getColorText() + "'" +
            "}";
    }
}

File: src/main/kotlin/com/patres/timetable/service/dto/SubjectDTO.kt
package com.patres.timetable.service.dto


import com.patres.timetable.service.dto.preference.PreferenceDataTimeForSubjectDTO
import com.patres.timetable.service.dto.preference.PreferenceSubjectByPlaceDTO
import java.io.Serializable
import java.util.HashSet
import javax.validation.constraints.NotNull

class SubjectDTO(

    @get:NotNull
    var name: String? = null,

    var shortName: String? = null,

    var colorBackground: String? = null,

    var colorText: String? = null,

    var preferenceSubjectByPlace: Set<PreferenceSubjectByPlaceDTO> = HashSet(),

    var preferencesDataTimeForSubject: Set<PreferenceDataTimeForSubjectDTO> = HashSet(),

    divisionOwnerId: Long? = null

) : AbstractDivisionOwnerDTO(divisionOwnerId = divisionOwnerId), Serializable
